pkg/endpoints/pipeline_test.go

```diff
@@ -1145,3 +1085,245 @@ func TestPipelineRunUpdateStatusAlreadySet412(t *testing.T) {
+			"reponame":        	 "testreponame"}`)
+
+	pipelineRunRequest := dummyHTTPRequest("POST", "http://wwww.dummy.com:8383/v1/namespaces/ns1/pipelinerun", pipelineRunBody)
+	pipelineRunRequestRestful := dummyRestfulRequest(pipelineRunRequest, "ns1", "")
+	resp := dummyRestfulResponse(httpWriter)
+	r.createPipelineRun(pipelineRunRequestRestful, resp)
+
+	pipelineResourceList, err := r.PipelineClient.TektonV1alpha1().PipelineResources("ns1").List(metav1.ListOptions{})
+
+	numberOfImageResources := 0
+
+	for _, pipelineResource := range pipelineResourceList.Items {
+		if pipelineResource.Spec.Type == "image" {
+			numberOfImageResources++
+		}
+	}
+
+	if numberOfImageResources != 1 {
+		t.Errorf("FAIL: expected to find a single created PipelineResource of type image, but found the number of image PipelineResources to be %d", numberOfImageResources)
+	}
+
+	t.Logf("Pipeline resource list: %v", pipelineResourceList)
+
+	if resp.StatusCode() != 204 {
+		t.Errorf("FAIL: should have been recognised as a 204, got %d", resp.StatusCode())
+	}
+}
+
+/* PipelineRun create with Git resource and image resource */
+func TestCreatePipelineRunGitAndImageResource(t *testing.T) {
+
+	r := dummyResource()
+
+	pipeline1 := v1alpha1.Pipeline{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "Pipeline1",
+		},
+		Spec: v1alpha1.PipelineSpec{},
+	}
+
+	_, err := r.PipelineClient.TektonV1alpha1().Pipelines("ns1").Create(&pipeline1)
+	if err != nil {
+		t.Errorf("FAIL: error creating test pipeline: %s", err)
+	}
+
+	httpWriter := httptest.NewRecorder()
+
+	pipelineRunBody := strings.NewReader(
+		`{"pipelinename":      "Pipeline1",
+			"imageresourcename": "image-source",
+			"gitresourcename":   "git-source",
+			"gitcommit":         "12345",
+			"reponame":          "testreponame",
+			"repourl":           "http://github.com/testorg/testrepo"}`)
+
+	pipelineRunRequest := dummyHTTPRequest("POST", "http://wwww.dummy.com:8383/v1/namespaces/ns1/pipelinerun", pipelineRunBody)
+	pipelineRunRequestRestful := dummyRestfulRequest(pipelineRunRequest, "ns1", "")
+	resp := dummyRestfulResponse(httpWriter)
+	r.createPipelineRun(pipelineRunRequestRestful, resp)
+
+	pipelineResourceList, err := r.PipelineClient.TektonV1alpha1().PipelineResources("ns1").List(metav1.ListOptions{})
+
+	numberOfGitResources := 0
+	numberOfImageResources := 0
+
+	resourceListingAsItems := pipelineResourceList.Items
+
+	for _, pipelineResource := range resourceListingAsItems {
+		if pipelineResource.Spec.Type == "git" {
+			numberOfGitResources++
+		} else if pipelineResource.Spec.Type == "image" {
+			numberOfImageResources++
+		}
+	}
+
+	if numberOfGitResources != 1 {
+		t.Errorf("FAIL: expected one created PipelineResource of type git but found the number of git PipelineResources to be %d", numberOfGitResources)
+	}
+
+	if numberOfImageResources != 1 {
+		t.Errorf("FAIL: expected one created PipelineResource of type image but found the number of image PipelineResources to be %d", numberOfImageResources)
+	}
```

Comment:
Same as the above comment about verifying content. You should be able to create functions that you can reuse here.